trino/auth.py

```diff
@@ -322,15 +323,18 @@ def _attempt_oauth(self, response, **kwargs):
         auth_info_headers = parse_dict_header(_OAuth2TokenBearer._BEARER_PREFIX.sub("", auth_info, count=1))
 
         auth_server = auth_info_headers.get('x_redirect_server')
-        if auth_server is None:
+        if auth_server is None and self._redirect_auth_url is None:
```

Comment:
I think we can remove this condition. If `x_redirect_server` is optional then it's optional. Previous value is not important.

Reply:
Yes, there is no need to check if `x_redirect_server` is present in this case. I've also updated tests cases that were failing because of the missing `x_redirect_server` in the response